Gemfile.lock

```diff
@@ -0,0 +1,97 @@
+PATH
+  remote: .
+  specs:
+    just-the-docs (0.5.4)
+      jekyll (>= 3.8.5)
+      jekyll-seo-tag (>= 2.0)
+      rake (>= 12.3.1)
+
+GEM
+  remote: https://rubygems.org/
+  specs:
+    addressable (2.8.4)
+      public_suffix (>= 2.0.2, < 6.0)
+    colorator (1.1.0)
+    concurrent-ruby (1.2.2)
+    em-websocket (0.5.3)
+      eventmachine (>= 0.12.9)
+      http_parser.rb (~> 0)
+    eventmachine (1.2.7)
+    faraday (2.7.10)
+      faraday-net_http (>= 2.0, < 3.1)
+      ruby2_keywords (>= 0.0.4)
+    faraday-net_http (3.0.2)
+    ffi (1.15.5)
+    forwardable-extended (2.6.0)
+    google-protobuf (3.23.4-x86_64-linux)
+    http_parser.rb (0.8.0)
+    i18n (1.14.1)
+      concurrent-ruby (~> 1.0)
+    jekyll (4.3.2)
+      addressable (~> 2.4)
+      colorator (~> 1.0)
+      em-websocket (~> 0.5)
+      i18n (~> 1.0)
+      jekyll-sass-converter (>= 2.0, < 4.0)
+      jekyll-watch (~> 2.0)
+      kramdown (~> 2.3, >= 2.3.1)
+      kramdown-parser-gfm (~> 1.0)
+      liquid (~> 4.0)
+      mercenary (>= 0.3.6, < 0.5)
+      pathutil (~> 0.9)
+      rouge (>= 3.0, < 5.0)
+      safe_yaml (~> 1.0)
+      terminal-table (>= 1.8, < 4.0)
+      webrick (~> 1.7)
+    jekyll-github-metadata (2.16.0)
+      jekyll (>= 3.4, < 5.0)
+      octokit (>= 4, < 7, != 4.4.0)
+    jekyll-sass-converter (3.0.0)
+      sass-embedded (~> 1.54)
+    jekyll-seo-tag (2.8.0)
+      jekyll (>= 3.8, < 5.0)
+    jekyll-watch (2.2.1)
+      listen (~> 3.0)
+    kramdown (2.4.0)
+      rexml
+    kramdown-parser-gfm (1.1.0)
+      kramdown (~> 2.0)
+    liquid (4.0.4)
+    listen (3.8.0)
+      rb-fsevent (~> 0.10, >= 0.10.3)
+      rb-inotify (~> 0.9, >= 0.9.10)
+    mercenary (0.4.0)
+    octokit (6.1.1)
+      faraday (>= 1, < 3)
+      sawyer (~> 0.9)
+    pathutil (0.16.2)
+      forwardable-extended (~> 2.6)
+    public_suffix (5.0.3)
+    rake (13.0.6)
+    rb-fsevent (0.11.2)
+    rb-inotify (0.10.1)
+      ffi (~> 1.0)
+    rexml (3.2.6)
+    rouge (4.1.2)
+    ruby2_keywords (0.0.5)
+    safe_yaml (1.0.5)
+    sass-embedded (1.64.1-x86_64-linux-gnu)
+      google-protobuf (~> 3.23)
```

Comment:
```suggestion
    sass-embedded (1.64.1-arm64-darwin)
      google-protobuf (~> 3.23)
    sass-embedded (1.64.1-x86_64-linux)
      google-protobuf (~> 3.23)
```

Reply:
I need to test this one first - the `linux-gnu`-thing again.

Reply:
```text
$  bundle install                                                                                                                                                                    
Fetching gem metadata from https://rubygems.org/...........
Your bundle is locked to sass-embedded (1.64.1-arm64-darwin) from rubygems repository https://rubygems.org/ or installed locally, but that version can no longer be found in that
source. That means the author of sass-embedded (1.64.1-arm64-darwin) has removed it. You'll need to update your bundle to a version other than sass-embedded (1.64.1-arm64-darwin) that
hasn't been removed in order to install.```

Reply:
Oh I see - the change in the bundler version is causing this discrepancy. Are you testing this locally with Ruby 3.2? 

(ideally, I'd like to stick on the version of bundler that syncs with our CI ruby versions, which is 3.1 That would be `2.3.26`. You would get a version that is quite similar to #1112)

Reply:
The CI gets the bundler version that was used to create the Lockfile.

Reply:
You can test the current Version of the lockfile on your machine. If a `bundle install` works and doesn't cause changes to the lockfile, I'm fine taking it in.

Reply:
> The CI gets the bundler version that was used to create the Lockfile.

This is usually not the norm (and to my understanding, is not how Actions works); unless you specify a bundler version, the default one packaged with that version of Ruby is used. I think this is generally a good idea because it matches what our users will most likely do, and makes sure that they don't get warnings when first building the site.

For what it's worth, after [ec5bdd3](https://github.com/just-the-docs/just-the-docs/pull/1307/commits/ec5bdd3e717313fa79ef55b1cf67cdd1c13c9062), I still don't get a clean `bundle install`; it resets the platform for `sass-embedded`

```
    sass-embedded (1.64.1-x86_64-linux)
```

which, I believe is partially related to bundler 2.4's [new resolution algorithm](https://bundler.io/blog/2023/01/31/bundler-v2-4.html).

Reply:
Alright. My change works for you, but your change doesn't work for me. According to rubygems.org, my version is correct.

We should change the pipeline to match the bundler version used to generate the lockfile. The lockfile is proof the used bundler version works.

Reply:
Thanks for your patience, catching up to all my JtD obligations.

> Alright. My change works for you, but your change doesn't work for me. According to rubygems.org, my version is correct.
> 
> We should change the pipeline to match the bundler version used to generate the lockfile. The lockfile is proof the used bundler version works.

Respectfully, I disagree. Our [gemspec](https://github.com/just-the-docs/just-the-docs/blob/7602fa1eae015fb51ea7a7d0aaecdc05d3ad00f8/just-the-docs.gemspec#L22) explicitly places a lower bound on the bundler version to be at `2.3`. More broadly, it's still not clear to me what the "repeating CI issues" are. I don't have an issue with updating our Ruby versions (ex bumping to `3.2`, pushing up the bundler dependency, etc.), but in that case we should be consistent across our repository (and thus, within this PR).

To be frank, I would have preferred that we continued off of the existing conversation and work in #1112. However, I am wary of bikeshedding / spreading our resources too thin, and at the end of the day this is something that we're all volunteering our time on. There is nothing objectively wrong with this PR - just suggestions that I have - and thus, if you're adamant that this current state is what should get merged in, I can give an approving review. Let me know what you think!